examples/tutorials/advanced/legends.py

```diff
@@ -0,0 +1,141 @@
+as manually created legends are supported.
+"""
+
+# %%
+import io
+
+import pygmt
+
+# %%
+# Create an auto-legend
+# ---------------------
+#
+# For auto-legends, the ``label`` parameter of :meth:`pygmt.Figure.plot` has to be
+# specified to state the desired text in the legend entry (white spaces are allowed).
+# Optionally, to adjust the legend, users can append different modifiers to the string
+# passed to ``label``. A list of all available modifiers can be found at
+# :gmt-docs:`gmt.html#l-full`. To create a
+# :doc:`multiple-column legend </gallery/embellishments/legend>` **+N** is used with
+# the desired number of columns. By default, the legend is placed in the Upper Right
+# corner with an offset of 0.1 centimeters in both x and y directions and a box with
+# a white fill and a 1-point thick, black, solid outline is drawn around the legend.
+
+fig = pygmt.Figure()
+fig.basemap(region=[-5, 5, -5, 5], projection="X5c", frame=True)
+
+fig.plot(x=0, y=0, style="c0.25c", fill="orange", label="orange circle")
+fig.legend()
+
+fig.show()
+
+
+# %%
+# Adjust the position
+# -------------------
+# Use the ``position`` parameter to adjust the position of the legend. For the
+# different ways to specify the placement of a plotting element (e.g., legends,
+# colorbars) on a plot in GMT, please refer to the Technical Reference TODO (: .
+# Add an offset via **+o** for the x and y directions. Additionally append **+w**
+# to adjust the ``width`` of the length. Note, no box is drawn by default if
+# ``position`` is used.
+
+fig = pygmt.Figure()
+fig.basemap(region=[-5, 5, -5, 5], projection="X5c", frame=True)
+
+fig.plot(x=0, y=0, style="c0.25c", fill="orange", label="orange circle")
+fig.legend(position="jLT+o0.3c/0.2c")
+
+fig.show()
+
+
+# %%
+# Add a box
+# ---------
+# Use the ``box`` parameter for adjusting the box around the legend. Append **+g**
+# to fill the legend with a color (or pattern) [Default is a white fill]. The
+# outline of the box can be adjusted by appending **+p**. The default of
+# ``position`` is preserved.
+
+
+fig = pygmt.Figure()
+fig.basemap(region=[-5, 5, -5, 5], projection="X5c", frame=True)
+
+fig.plot(x=0, y=0, style="c0.25c", fill="orange", label="orange circle")
+fig.legend(position="jTL+o0.3c/0.2c", box=True)
+
+fig.shift_origin(xshift="w+1c")
+fig.basemap(region=[-5, 5, -5, 5], projection="X5c", frame=True)
+
+fig.plot(x=0, y=0, style="c0.25c", fill="orange", label="orange circle")
+fig.legend(position="jTL+o0.3c/0.2c", box="+p2p,cyan+gblue@70")
+
+fig.show()
+
+
+# %%
+# Create a manual legend
+# ----------------------
+#
+# For more complicated legends in GMT, users need to write an ASCII file with
+# instructions for the layout of the legend items. In PyGMT it is additionally
+# possible to provide this information as an :class:`io.StringIO` object. Both, the
+# ASCII file or the :class:`io.StringIO` object are passed to the ``spec`` parameter
+# of :meth:`pygmt.Figure.legend`.
+#
```

Comment:
Also need to add a link to https://docs.generic-mapping-tools.org/dev/legend.html#legend-codes.

Reply:
Yes, this part needs some work on the documentation side. Will do this in the next days 🙂.